internal/usecase/interactor/auth.go

```diff
@@ -57,7 +59,7 @@ var dummyName = pkix.Name{
 	PostalCode:         []string{"1"},
 }
 
-func NewAuthStorage(cfg *StorageConfig, request repo.AuthRequest, getUserBySubject func(context.Context, string) (*user.User, error)) op.Storage {
+func NewAuthStorage(ctx context.Context, cfg *StorageConfig, request repo.AuthRequest, config repo.Config, getUserBySubject func(context.Context, string) (*user.User, error)) op.Storage {
```

Comment:
```suggestion
func NewAuth(ctx context.Context, cfg *StorageConfig, request repo.AuthRequest, config repo.Config, getUserBySubject func(context.Context, string) (*user.User, error)) (op.Storage, error) {
```

Panic should be caused in one function; other functions should return errors, because we may want to change the behavior to avoid panicking, and if the panic calls are littered with them, it can be an obstacle to troubleshooting.